src/analysis/processing/qgsalgorithmcellstatistics.cpp

```diff
@@ -0,0 +1,333 @@
+}
+
+bool QgsCellStatisticsAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * feedback )
+{
+  QgsRasterLayer *referenceLayer = parameterAsRasterLayer( parameters, QStringLiteral( "REF_LAYER" ), context );
+  if ( !referenceLayer )
+    throw QgsProcessingException( invalidRasterError( parameters, QStringLiteral( "REF_LAYER" ) ) );
+
+  mIgnoreNoData = parameterAsBool( parameters, QStringLiteral( "IGNORE_NODATA" ), context);
+
+  mCrs = referenceLayer->crs();
+  mRasterUnitsPerPixelX = referenceLayer->rasterUnitsPerPixelX();
+  mRasterUnitsPerPixelY = referenceLayer->rasterUnitsPerPixelY();
+  mLayerWidth = referenceLayer->width();
+  mLayerHeight = referenceLayer->height();
+  mExtent = referenceLayer->extent();
+
+  const QList< QgsMapLayer * > layers = parameterAsLayerList( parameters, QStringLiteral( "INPUTS" ), context );
+  QList< QgsRasterLayer * > rasterLayers;
+  rasterLayers.reserve( layers.count() );
+  for ( QgsMapLayer *l : layers )
+  {
+    if(feedback->isCanceled())
+      break; //in case some slow data sources are loaded
+
+    if ( l->type() == QgsMapLayerType::RasterLayer )
+    {
+      QgsRasterLayer *layer = qobject_cast< QgsRasterLayer * >( l );
+      QgsRasterAnalysisUtils::RasterLogicInput input;
+      const int band = 1; //could be made dynamic
+      input.hasNoDataValue = layer->dataProvider()->sourceHasNoDataValue( band );
+      input.sourceDataProvider.reset( layer->dataProvider()->clone() );
+      input.interface = input.sourceDataProvider.get();
+      // add projector if necessary
+      if ( layer->crs() != mCrs )
+      {
+        input.projector = qgis::make_unique< QgsRasterProjector >();
+        input.projector->setInput( input.sourceDataProvider.get() );
+        input.projector->setCrs( layer->crs(), mCrs, context.transformContext() );
+        input.interface = input.projector.get();
+      }
+      mInputs.emplace_back( std::move( input ) );
+    }
+  }
+
+  return true;
+}
+
+QVariantMap QgsCellStatisticsAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  //obtain statistic method
+  int statisticMethodIdx = parameterAsInt( parameters, QStringLiteral( "STATISTIC" ), context );
+
+  //determine output raster data type
+  //initially raster data type to most primitive data type that is possible
+  mDataType = Qgis::Byte;
+  for ( const QgsRasterAnalysisUtils::RasterLogicInput &i : mInputs )
+  {
+    for ( int band : i.bands )
+    {
+      Qgis::DataType inputDataType = i.interface->dataType( band );
+      if( static_cast<int>(mDataType) < static_cast<int>(inputDataType) )
+        mDataType = inputDataType; //if raster data type is more potent, set it as new data type
+    }
+  }
+
+  //force data types on specific functions if input data types don't match
+  if( statisticMethodIdx == 2 || statisticMethodIdx == 4 ) //mean, stddev
```

Comment:
Would be cleaner to make an enum for the statistic calculation types

Reply:
Implemented